### PR TITLE
Retry when checking splits in Cassandra tests

### PR DESCRIPTION
**Pull request by sanjay990**

### CODE REVIEW

**Comment by brian-rickman** [comment]
I am inclined to agree with @maciejgrzybek .  This should be fixed in getSplits rather than in the test.

File: presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraTokenSplitManager.java [hunk 1]
```diff
@@ -65,8 +69,25 @@ public void testNonEmptyTable()
...
+    private void checkSplits(String tableName, int splitSize)
+    {
+        long start = System.nanoTime();
+        while (!Thread.currentThread().isInterrupted()) {
+            try {
+                List<TokenSplit> splits = splitManager.getSplits(KEYSPACE, tableName);
+                assertEquals(splits.size(), splitSize);
+                return;
+            }
+            catch (AssertionError e) {
```
Comment (electrum) [inline presto-cassandra/src/test/java/com/facebook/presto/cassandra/TestCassandraTokenSplitManager.java]: I agree with the other comments, but wanted to point out that catching an exception that you explicitly throw (via `assertEquals`) should be avoided. This retry loop could be written to loop until time expires or the count correct.

**Comment by cawallin** [comment]
I actually don't think this needs to be changed in `CassandraSplitManager`. The retry is needed because Cassandra's consistency model by default does not guarantee that tables/stats are visible after the `CREATE TABLE` has completed. `CassandraSplitManager` should be fine since it returns the tables/stats as of when it is querying, which means that Presto obeys the consistency model of Cassandra (also, how would you know that the stats for a particular table are out of date? e.g. how would you know when to stop retrying?). This test fix could be clearer if a retry were added to `EmbeddedCassandra.refreshSizeEstimates(KEYSPACE, tableName);` instead.

**Comment by kokosing** [comment]
@sanjay990 I am going to close this since there are unaddressed comments (blockers) and it seems that it is no longer needed. Please reopen if I am wrong.

**Pull request closed without merging**